components/provisioning/org.wso2.carbon.identity.provisioning/src/main/java/org/wso2/carbon/identity/provisioning/OutboundProvisioningManager.java

```diff
@@ -121,75 +123,86 @@ private static int getTenantIdOfDomain(String tenantDomain)
     private Map<String, RuntimeProvisioningConfig> getOutboundProvisioningConnectors(
             ServiceProvider serviceProvider, String tenantDomainName) throws IdentityProvisioningException {
 
+        if (serviceProvider == null) {
+            throw new IdentityProvisioningException("Given Service Provider is null hence Outbound Provisioning " +
+                    "Connectors cannot be retrieved for tenant:" + tenantDomainName);
+        }
+
+        if (CarbonContext.getThreadLocalCarbonContext() == null) {
+            throw new IdentityProvisioningException("ThreadLocalCarbonContext cannot be null");
+        }
+
         Map<String, RuntimeProvisioningConfig> connectors = new HashMap<>();
+        String tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        int tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
 
-        // maintain the provisioning connector cache in the super tenant.
-        // at the time of provisioning there may not be an authenticated user in the system -
-        // specially in the case of in-bound provisioning.
 
-        String tenantDomain = null;
-        int tenantId = -1234;
-        ServiceProviderProvisioningConnectorCacheKey key = null;
-        ServiceProviderProvisioningConnectorCacheEntry entry = null;
+        connectors = getRuntimeProvisioningConfigFromCache(serviceProvider,
+                tenantDomainName, tenantDomain, tenantId);
+        if (!connectors.isEmpty()) {
+            return connectors;
+        }
 
-        if (CarbonContext.getThreadLocalCarbonContext() != null) {
-            tenantDomain = CarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-            tenantId = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        connectors = getRuntimeProvisioningConfig(serviceProvider, tenantDomainName);
+        if (!connectors.isEmpty()) {
+            addToCache(serviceProvider, connectors, tenantDomain, tenantId);
+            if (log.isDebugEnabled()) {
+                log.debug("Entry added successfully ");
```

Comment:
This message is not helpful in debuting, what is this "Entry",  successfully  added to what ?   Please provide more context in debug logs